docs/fud/synthesis.md

```diff
@@ -1,44 +1,124 @@
-
-## Vivado and Vivado HLS over SSH
-`fud` supports invoking these tools over SSH. You have to tell `fud` the username and hostname
-for a server that has these tools installed:
-```
-# vivado
-fud config stages.synth-verilog.ssh_host <hostname>
-fud config stages.synth-verilog.ssh_username <username>
-
-# vivado hls
-fud config stages.vivado-hls.ssh_host <hostname>
-fud config stages.vivado-hls.ssh_username <username>
-```
-
-**Note:** `vivado` or `vivado_hls` have to be on the path of the remote machine
-for this to work.
-If you need the names to be something else, file an issue.
-`fud` currently does not support other names.
-
-## Vivado/VivadoHLS locally
-We don't provide installation instructions for this. However, `fud` will look for
-`vivado` and `vivado-hls` binaries on the system. If these are installed, you can
-use `fud` to invoke these tools. You can change the paths `fud` looks for with
-```
-fud config stages.synth-verilog.exec <path> # update vivado path
-fud config stages.vivado-hls.exec <path> # update vivado_hls path
-```
+# Xilinx Toolchain
 
+> Working with vendor EDA toolchains is never a fun experience. Something will
+> almost certainly go wrong. If you're at Cornell, you can at least avoid
+> installing the tools yourself by using our lab servers, [Gorgonzola][] or [Havarti][].
+
+`fud` can interact with the Xilinx tools ([Vivado][], [Vivado HLS][vhls], and [Vitis][]). There are three main things it can do:
+
+* Synthesize Calyx-generated RTL designs to collect area and resource estimates.
+* Compile [Dahlia][] programs via C++ and [Vivado HLS][vhls] for comparison with the Calyx backend.
+* Compile Calyx programs for *actual execution* in Xilinx emulation modes or on real FPGA hardware.
+
+You can set `fud` up to use either a local installation of the Xilinx tools or one on a remote server, via SSH.
+
+## Synthesis Only
+
+The simplest way to use the Xilinx tools is to synthesize RTL or HLS designs to collect statistics about them.
+This route will not produce actual, runnable executables; see the next section for that.
+
+### Set Up
+
+To set up to **invoke the Xilinx tools over SSH**, first tell `fud` your username and hostname for the server:
+
+    # Vivado
+    fud config stages.synth-verilog.ssh_host <hostname>
+    fud config stages.synth-verilog.ssh_username <username>
+
+    # Vivado HLS
+    fud config stages.vivado-hls.ssh_host <hostname>
+    fud config stages.vivado-hls.ssh_username <username>
+
+The server must have `vivado` and `vivado_hls` available on the remote machine's path. (If you need the executable names to be something else, please file an issue.)
+
+To instead **invoke the Xilinx tools locally**, just let `fud` run the `vivado` and `vivado_hls` commands.
+You can optionally tell `fud` where these commands exist on your machine:
+
+    fud config stages.synth-verilog.exec <path> # update vivado path
+    fud config stages.vivado-hls.exec <path> # update vivado_hls path
```

Comment:
What happens if both `ssh_host` stuff and `exec` path here are set in the configuration? Does it default to local execution? Is there a way to force execution on either when both are provided?

Reply:
Good question; I clarified that there is actually a `remote` flag involved to decide which option to use.

One important but small to-do item for the future is centralizing the config so there is only one set of `ssh_host`, etc., options instead of one per stage, which quickly gets tiresome…

Reply:
Yup, but doing that kind of configuration sharing is kind of annoying. See https://github.com/cucapra/calyx/issues/751

Reply:
Indeed, it will take a bit of effort…